tests/auth0/handlers/migrations.tests.js

```diff
@@ -0,0 +1,112 @@
+const { expect } = require('chai');
+const migrations = require('../../../src/auth0/handlers/migrations');
+
+describe('#migrations handler', () => {
+  const mockClient = flags => ({
+    migrations: {
+      getMigrations: () => ({
+        flags: {
+          migration_flag: true
+        }
+      }),
+      updateMigrations: (data) => {
+        expect(data).to.be.an('object');
+        expect(data).to.have.deep.property('flags', { migration_flag: false, ...flags });
+        return Promise.resolve(data);
+      }
+    }
+  });
+
+  describe('#getType()', () => {
+    it('should get migration flags', async () => {
+      const client = mockClient();
+
+      const handler = new migrations.default({ client });
+      const data = await handler.getType();
+      expect(data).to.deep.equal({
+        migration_flag: true
+      });
+    });
+  });
+
+  describe('#migrations process', () => {
+    it('should update available migration flags', async () => {
+      const client = mockClient();
+      const config = () => false;
+
+      const handler = new migrations.default({ client, config });
+      const stageFn = Object.getPrototypeOf(handler).processChanges;
+
+      await stageFn.apply(handler, [ {
+        migrations: {
+          migration_flag: false
+        }
+      } ]);
```

Comment:
I'd use `Function.call` to avoid the extra array.

```suggestion
      await stageFn.call(handler, {
        migrations: {
          migration_flag: false
        }
      });
```

What is preventing you from using `handler.processChanges`? 🤔

Reply:
I don't think there is anything preventing it but instead the desire to mimic the actual call made by the main class https://github.com/auth0-extensions/auth0-source-control-extension-tools/blob/master/src/auth0/index.js#L36-L40

I can clean up to simplify but I was just following the convention of the repo instead of adding a new one.

Reply:
Fair enough, if that is the convention then let's not change it in this PR.